File: backend/app/schemas/admin_qc.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class QualityIssuePublic(BaseModel):
    id: uuid.UUID
    dataset_id: uuid.UUID
    dataset_title: str
    issue_type: str
    severity: str
    status: str
    detail: str | None
    detected_at: datetime

    model_config = {"from_attributes": True}


class QcScanResult(BaseModel):
    issues_found: int
    issues: list[QualityIssuePublic]


class QualityIssueStatusUpdate(BaseModel):
    status: Literal["resolved", "ignored"]

File: frontend/src/lib/types/admin-qc.ts
// Mirrors backend/app/schemas/admin_qc.py exactly — keep these two in sync.

export interface QualityIssuePublic {
  id: string;
  dataset_id: string;
  dataset_title: string;
  issue_type: string;
  severity: string;
  status: string;
  detail: string | null;
  detected_at: string;
}

export interface QcScanResult {
  issues_found: number;
  issues: QualityIssuePublic[];
}

export type QualityIssueStatus = "resolved" | "ignored";

export interface QualityIssueStatusUpdate {
  status: QualityIssueStatus;
}
